src/core/dependenciesMintPolicy.js

```diff
@@ -0,0 +1,133 @@
+ * the dependencies have a total weight of 0.2, then the total Cred will be
+ * 120% of the base Cred, but if the dependencies had a total weight of 1, then
+ * the total Cred would be double the base Cred. This was a deliberate design
+ * decision so that dependency minting would feel "non-rival", i.e. there is
+ * not a fixed budget of dependency cred that must be split between the
+ * dependencies. In many cases, it may be very reasonable for the total Cred
+ * flowing to a project's dependencies to be larger than the total Cred flowing
+ * directly to the project's contributors; consider that the total amount of
+ * time/effort invested in building all the dependencies may be orders of
+ * magnitude larger than investment in the project itself.
+ *
+ * In the future, we may develop more elegant approaches that integrate
+ * dependencies more naturally into the contribution graph itself, in which
+ * case we could phase out this system. But for now, it seems like a
+ * sufficiently useful and flexible abstraction to merit inclusion in core/.
+ *
+ * Note that there's nothing in the implementation or data types that imply
+ * that this is only used for dependencies; one could imagine using it for
+ * other purposes, e.g. minting extra Cred for moderators of a forum, or for
+ * maintainers of a project. However, we have a prior that all contributors to
+ * a project should be paid via CredRank and the graph-based system, rather
+ * than "out-of-band" fiat/configuration like the mint policies in this module.
+ * Thus, we're naming it the dependencyMintPolicy to communicate that
+ * intention.
+ */
+import {type NodeAddressT, NodeAddress} from "./graph";
+import {type TimestampMs} from "../util/timestamp";
+
+export type DependencyMintPolicy = {|
+  // The node address that will receieve the extra minted Cred
+  +address: NodeAddressT,
+  // Information on how the minted Cred varies in time.
+  +periods: $ReadOnlyArray<DependencyMintPeriod>,
+|};
+
+export type DependencyMintPeriod = {|
+  // What proportion of the project's raw Cred should be minted to this
+  // dependency during this period. For example, if the weight is 0.05, and the
+  // project has a pre-dependencies total Cred of 1000 in a given interval,
+  // then this dependench will receive an additional 50 Cred.
+  +weight: number,
+  // Timestamp or number because we allow -Infinity as the first "timestamp"
+  // Within any policy, the period timestamps must be in sorted order, or else
+  // an error will be thrown.
+  +startTimeMs: TimestampMs | number,
+|};
+
+/**
+ * The ProcessedMintPolicy is a DependencyMintPolicy which has
+ * been transformed so that it matches the abstractions available when
+ * we're doing raw cred computation: instead of an address, we track an index
+ * into the canonical node order, and rather than arbitrary client-provided
+ * periods, we compute the weight for each Interval.
+ */
+export type ProcessedMintPolicy = {|
```

Comment:
Should be named "ProcessedDependencyMintPolicy" IMO

Reply:
Changed.